xarray/core/datatree.py

```diff
@@ -949,7 +944,7 @@ def update(self, other: Dataset | Mapping[str, DataTree | DataArray]) -> None:
 
         vars_merge_result = dataset_update_method(self.to_dataset(), new_variables)
         # TODO are there any subtleties with preserving order of children like this?
-        merged_children = OrderedDict({**self.children, **new_children})
+        merged_children = dict({**self.children, **new_children})
```

Comment:
Another wrapper that can be removed probably.
```suggestion
        merged_children = {**self.children, **new_children}
```